src/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -1394,42 +1394,22 @@ protected void setAttribute(final short tag, final Object value) {
      * @param value the value to be checked
      * @return true if the value is valid and false otherwise
      */
+    @Deprecated
```

Comment:
could you add an `* @deprecated ` block to the javadoc explaining that it's deprecated because it moved into `SAMBinaryTagAndValue.isAllowedAttributeValue`  It's easy to figure out it out, but  it's even easier to have it tell you that it's safe to use the new method.